.github/workflows/pr.yaml

```
@@ -0,0 +1,327 @@
+name: Galaxy Tool Linting and Tests for push and PR
+on: [push, pull_request]
+env:
+  GALAXY_REPO: https://github.com/galaxyproject/galaxy
+  GALAXY_RELEASE: release_20.01
+jobs:
+  setup:
+    name: Setup cache and determine changed repositories
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+    steps:
+    - name: ECHO
+      run: echo event ${{ github.event_name }} sha ${{ github.sha }} ref ${{ github.ref }} ${{ github.event_name }} 
+    - name: ECHO
+      run: echo head_ref ${{ github.head_ref }} base_ref ${{ github.head_ref }} ${{ github.event.pull_request.head.sha }}
+    - uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache .cache/pip
+      uses: actions/cache@v1
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_py${{ matrix.python-version }}_${{ env.GALAXY_RELEASE }}
+    - name: Determine latest galaxy commit
+      run: echo ::set-env name=GALAXY_HEAD_SHA::$(git ls-remote ${{ env.GALAXY_REPO }} refs/heads/${{ env.GALAXY_RELEASE }} | cut -f1)
+    - name: Cache .planemo
+      uses: actions/cache@v1
+      id: cache-planemo
+      with:
+        path: ~/.planemo
+        key: planemo_cache_py${{ matrix.python-version }}_${{ env.GALAXY_HEAD_SHA }}
+    - name: Install Planemo and flake8
+      run: pip install planemo flake8
+    - name: Fake a planemo run to update cache
+      if: steps.cache-pip.outputs.cache-hit != 'true' || steps.cache-planemo.outputs.cache-hit != 'true'
+      run: |
+        touch tool.xml
+        PIP_QUIET=1 planemo test --galaxy_python_version ${{ matrix.python-version }} --no_conda_auto_init --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: get all branches
+      run: |
+        git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+    - name: Determine commit range for feature branch
+      if: github.ref != 'refs/heads/master'
+      run: echo ::set-env name=COMMIT_RANGE::"origin/master.."
```

I think this should be (as in `.travis.yml`):

```suggestion
      run: echo ::set-env name=COMMIT_RANGE::"origin/master..."
```

Otherwise the commit range may contain changes from `origin/master` that are not in the feature branch.

---

I think its the other way around: 

>  triple-dot syntax, which specifies all the commits that are reachable by either of two references but not by both of them

https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection

But I was also wondering, because in the travis config the tripple dot was used

---

The odd thing is that with `git log origin/master.. --oneline` I see exactly the commits that I want to (while `git log origin/master... --oneline ` I also see new commits from the master branch), but in `planemo ci_find_repos` `origin/master..` gives me nothing in some cases .. don't get it yet.

---

Got it: I was reading the wrong docs

> However, "diff" is about comparing two endpoints, not ranges, and the range notations